sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -2614,5 +2640,264 @@ private void SetNameFieldsIfNull()
+
+                    List<ListBlobsIncludeItem> include = new List<ListBlobsIncludeItem>
+                    {
+                        ListBlobsIncludeItem.Deleted
+                    };
+
+                    if (async)
+                    {
+                        response = await BlobFileSystemRestClient.ListBlobHierarchySegmentAsync(
+                            delimiter: null,
+                            prefix: path,
+                            marker: continuation,
+                            maxResults: maxResults,
+                            include: null,
+                            timeout: null,
+                            cancellationToken: cancellationToken)
+                            .ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        response = BlobFileSystemRestClient.ListBlobHierarchySegment(
+                            delimiter: null,
+                            prefix: path,
+                            marker: continuation,
+                            maxResults: maxResults,
+                            include: null,
+                            timeout: null,
+                            cancellationToken: cancellationToken);
+                    }
+
+                    return Response.FromValue(
+                        response.ToPathDeletedSegment(),
+                        response.GetRawResponse());
+                }
+                catch (Exception ex)
+                {
+                    ClientConfiguration.Pipeline.LogException(ex);
+                    scope.Failed(ex);
+                    throw;
+                }
+                finally
+                {
+                    ClientConfiguration.Pipeline.LogMethodExit(nameof(DataLakeFileSystemClient));
+                    scope.Dispose();
+                }
+            }
+        }
+        #endregion Get Deleted Paths
+
+        #region Restore Path
+        /// <summary>
+        /// Restores a soft deleted path.
+        /// </summary>
+        /// <param name="deletedPath">
+        /// Required.  The path of the deleted path.
+        /// </param>
+        /// <param name="deletionId">
+        /// Required.  The deletion ID associated with the soft deleted path.
+        /// You can get soft deleted paths and their assocaited deletion IDs with <see cref="GetDeletedPathsAsync"/>.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{DataLakePathClient}"/> pointed at the newly
+        /// restored path.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<DataLakePathClient> RestorePath(
```

Comment:
should this be UndeletePath (to be consistent with UndeleteFileSystem)?

Reply:
fixed.